hera_cal/tests/test_chunker.py

```diff
@@ -0,0 +1,118 @@
+from ..data import DATA_PATH
+from .. import chunker
+from hera_qm.utils import apply_yaml_flags
+import numpy as np
+import sys
+
+
+def test_chunk_data_files(tmpdir):
+    # list of data files:
+    tmp_path = tmpdir.strpath
+    data_files = sorted(glob.glob(DATA_PATH + '/zen.2458044.*.uvh5'))
+    nfiles = len(data_files)
+    # form chunks with three samples.
+    for chunk in range(0, nfiles, 2):
+        output = tmp_path + f'/chunk.{chunk}.uvh5'
+        chunker.chunk_files(data_files, data_files[chunk], output, 2,
+                            polarizations=['ee'], spw_range=[0, 32],
+                            throw_away_flagged_ants=True, ant_flag_yaml=DATA_PATH + '/test_input/a_priori_flags_sample_noflags.yaml')
+
+    # test that chunked files contain identical data (when combined)
+    # to original combined list of files.
+    # load in chunks
+    chunks = sorted(glob.glob(tmp_path + '/chunk.*.uvh5'))
+    uvd = UVData()
+    uvd.read(chunks)
+    # load in original file
+    uvdo = UVData()
+    uvdo.read(data_files, freq_chans=range(32))
+    apply_yaml_flags(uvdo, DATA_PATH + '/test_input/a_priori_flags_sample_noflags.yaml', throw_away_flagged_ants=True,
+                     flag_freqs=False, flag_times=False, ant_indices_only=True)
+    assert np.all(np.isclose(uvdo.data_array, uvd.data_array))
+    assert np.all(np.isclose(uvdo.flag_array, uvd.flag_array))
+    assert np.all(np.isclose(uvdo.nsample_array, uvd.nsample_array))
+
+    # Repeate test with no spw_range or pols provided.
+    for chunk in range(0, nfiles, 2):
+        output = tmp_path + f'/chunk.{chunk}.uvh5'
+        chunker.chunk_files(data_files, data_files[chunk], output, 2,
+                            polarizations=None, spw_range=None, clobber=True,
+                            throw_away_flagged_ants=True, ant_flag_yaml=DATA_PATH + '/test_input/a_priori_flags_sample_noflags.yaml')
+
+    # test that chunked files contain identical data (when combined)
+    # to original combined list of files.
+    # load in chunks
+    chunks = sorted(glob.glob(tmp_path + '/chunk.*.uvh5'))
+    uvd = UVData()
+    uvd.read(chunks)
+    # load in original file
+    uvdo = UVData()
+    uvdo.read(data_files)
+    apply_yaml_flags(uvdo, DATA_PATH + '/test_input/a_priori_flags_sample_noflags.yaml', throw_away_flagged_ants=True,
+                     flag_freqs=False, flag_times=False, ant_indices_only=True)
+    assert np.all(np.isclose(uvdo.data_array, uvd.data_array))
+    assert np.all(np.isclose(uvdo.flag_array, uvd.flag_array))
+    assert np.all(np.isclose(uvdo.nsample_array, uvd.nsample_array))
+
+
+def test_chunk_cal_files(tmpdir):
+    # list of data files:
+    tmp_path = tmpdir.strpath
+    cal_files = sorted(glob.glob(DATA_PATH + '/test_input/*.abs.calfits_54x_only.part*'))
+    nfiles = len(cal_files)
+    # test ValueError
+    pytest.raises(ValueError, chunker.chunk_files, cal_files, cal_files[0], 'output', 2, spw_range=[0, 32], type='arglebargle')
```

Comment:
In the spirit of making tests more atomic, I prefer to test errors like these in a separate test. It's also generally a good idea to make sure that the message raised by the error matches the one you expect (just so that it's not inadvertently raising the wrong error). This can be done with the `match` argument to `pytest.raises`, or you can do something more rigorous like this:

```
with pytest.raises(ValueError) as cm:
    chunker.chunk_files(cal_files, cal_files[0], "output", 2, spw_range=[0, 32], type="arglebargle")
assert str(cm.value).startswith("Invalid type provided. Must be in ['data', 'gains'].")
```
I'm fine to merge it in as-is, just trying to spread the word on testing best practices 🙂

Reply:
Thanks for the tip. I'll follow this in future PRs.